src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,18 +3,47 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAXIMUM_CAPACITY = 10;
+
+    private K[] keys;
+    private V[] values;
+    private int currentSize;
+
+    public StorageImpl() {
+        this.keys = (K[]) new Object[MAXIMUM_CAPACITY];
+        this.values = (V[]) new Object[MAXIMUM_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        boolean isFound = false;
+        if (currentSize < MAXIMUM_CAPACITY) {
+            for (int i = 0; i < currentSize; i++) {
+                if (keys[i] == key || (keys[i] != null && keys[i].equals(key))) {
+                    values[i] = value;
+                    isFound = true;
+                }
```

Comment:
```suggestion
            for (int i = 0; i < currentSize; i++) {
                if (keys[i] == key || (keys[i] != null && keys[i].equals(key))) {
                    values[i] = value;
                    return;
                }
```
you don't need first if condition. If storage is full for user it's better to receive exception to know that they can't add anymore values. If not they may think that value was saved, but it wasn't

Reply:
f (currentSize < MAXIMUM_CAPACITY) replaced by ArrayIndexOutOfBoundsException